crates/core_simd/src/vector.rs

```diff
@@ -242,11 +254,10 @@ where
     #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
     pub unsafe fn to_int_unchecked<I>(self) -> Simd<I, LANES>
     where
-        T: core::convert::FloatToInt<I>,
-        I: SimdElement,
+        T: core::convert::FloatToInt<I> + SimdCast,
+        I: SimdCast,
```

Comment:
Huh, so this happens because `SimdElement` is now either `SimdCast` xor `SimdPtrCast`, right?

Reply:
Yeah.  But really, it has nothing to do with `SimdPtrCast` (if rust ever added a numeric pointer type for some reason, it could implement both).  This could be thought of as `SimdNumeric`, but specific to casting.

Reply:
Ahhh okay.

Reply:
I think I will change it to `T: SimdCast<U>`, to match the updated `SimdPtrCast` (and in case we want to use similar bounds in the future).

Reply:
Actually, I realized this would require a quadratic number of implementations... It's probably fine how it is.